src/main/java/net/fabricmc/mappingio/tree/MappingTree.java

```diff
@@ -157,36 +159,14 @@ default String mapDesc(CharSequence desc, int start, int end, int srcNamespace,
 		return ret.toString();
 	}
 
-	public interface ElementMapping {
+	interface ElementMapping extends ElementMappingView {
 		MappingTree getTree();
 
-		String getSrcName();
-		String getDstName(int namespace);
-
-		default String getName(int namespace) {
-			if (namespace < 0) {
-				return getSrcName();
-			} else {
-				return getDstName(namespace);
-			}
-		}
-
-		default String getName(String namespace) {
-			int nsId = getTree().getNamespaceId(namespace);
-
-			if (nsId == NULL_NAMESPACE_ID) {
-				return null;
-			} else {
-				return getName(nsId);
-			}
-		}
-
 		void setDstName(int namespace, String name);
-		String getComment();
 		void setComment(String comment);
 	}
 
-	public interface ClassMapping extends ElementMapping {
+	interface ClassMapping extends ClassMappingView {
```

Comment:
```suggestion
	interface ClassMapping extends ElementMapping, ClassMappingView {
```
same for other interfaces that additionally implement a `View` insterface. Then you can remove those spurious casts.

Reply:
fixed